lib/Frontend/ParseableOutput.cpp

```diff
@@ -0,0 +1,244 @@
+//===--- ParseableOutput.cpp - Helpers for parseable output ---------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2017 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+#include "swift/Frontend/ParseableOutput.h"
+
+#include "swift/Basic/FileTypes.h"
+#include "swift/Basic/JSONSerialization.h"
+#include "swift/Basic/TaskQueue.h"
+#include "llvm/Option/Arg.h"
+#include "llvm/Support/raw_ostream.h"
+
+using namespace swift::parseable_output;
+using namespace swift::sys;
+using namespace swift;
+
+namespace swift {
```

Comment:
Some sort of overall comment would be nice.
Are all these traits teaching a JSON formatter how to output this stuff? (That's my sleepy guess.)

Reply:
Yes, this is why these traits are defined in the `json` namespace

Reply:
If you needed to extend the parsable output in a way that involved these traits, and you were brand-new to the code, would a comment here help? Or do you think it's just obvious?